### PR TITLE
feat: show call trace on transaction revert

### PR DESCRIPTION
**Pull request by mikeshultz**
### What I did

Shows call trace on transaction revert.

Ref: https://github.com/ApeWorX/ape/issues/1945

### Checklist

- [X] Passes all linting checks (pre-commit and CI jobs)
- [ ] New test cases have been added and are passing
- [ ] Documentation has been updated
- [X] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

File: ape_hardhat/provider.py [hunk 1]
```diff
@@ -1001,45 +1001,51 @@ def get_virtual_machine_error(self, exception: Exception, **kwargs) -> VirtualMa
...
+            )
+
+            if enriched.message == TransactionError.DEFAULT_MESSAGE and revert_message:
+                # Since input data is always missing, and to preserve backwards compat,
+                # use the selector as the message still.
+                enriched.message = revert_message
+
+            # Show call trace if availble
+            if enriched.txn:
+                # Apparently this is not always a receipt on a failed tx
```
Comment (antazoey) [inline ape_hardhat/provider.py:1019]: Correct, but likely if that is the case, tx.receipt will be None. For example, if your tx fails during estimation, there won't be a receipt.

**Review by antazoey — APPROVED** [review]
Looks good, but see my comment.. you can probably just ignore the case when there isnt a receipt and only a tx

**Comment by antazoey** [comment]
Or maybe we will have to show the trace a different way in this case.

It is ok if it is ugly for now, there is a branch for 0.8 trace refactor we can bypass this a lot easier.

File: ape_hardhat/provider.py (hunk 1 above)
Comment (mikeshultz) [reply ape_hardhat/provider.py:1019]: It does check if the receipt exists before attempting to show the trace.  Mostly just covering all the possibilities here.

**Review by antazoey — APPROVED** [review]

**Pull request merged**